scalalib/src/mill/javalib/android/AndroidAppModule.scala

```diff
@@ -452,4 +512,230 @@ trait AndroidAppModule extends JavaModule {
     PathRef(Task.dest)
   }
 
+  /** The name of the virtual device to be created by  [[createAndroidVirtualDevice]] */
+  def virtualDeviceIdentifier: String = "test"
+  private def deviceIdentifier = virtualDeviceIdentifier
+
+  /**
+   * The target architecture of the virtual device to be created by  [[createAndroidVirtualDevice]]
+   *  For example, "x86_64" (default). For a list of system images and their architectures,
+   *  see the Android SDK Manager `sdkmanager --list`.
+   */
+  def emulatorArchitecture: String = "x86_64"
+  private def arch = emulatorArchitecture
+
+  /**
+   * Installs the user specified system image for the emulator
+   * using sdkmanager . E.g. "system-images;android-35;google_apis_playstore;x86_64"
+   */
+  def sdkInstallSystemImage: Target[String] = Task {
+    val image =
+      s"system-images;${androidSdkModule().platformsVersion()};google_apis_playstore;${emulatorArchitecture}"
+    os.call((
+      androidSdkModule().sdkManagerPath().path,
+      "--install",
+      image
+    ))
+    image
+  }
+
+  /**
+   * Creates the android virtual device identified in virtualDeviceIdentifier
+   */
+  def createAndroidVirtualDevice: T[os.CommandResult] = Task {
+    os.call((
+      androidSdkModule().avdPath().path,
+      "create",
+      "avd",
+      "--name",
+      virtualDeviceIdentifier,
+      "--package",
+      sdkInstallSystemImage(),
+      "--device",
+      "Nexus 5X",
```

Comment:
this is a pretty old device definition to use by default. Any particular reason to use this one and not recent Pixel definitions which bring more actual parameters like screen size?

Reply:
ah thanks for spotting, I was trying to use a lower end device for running 2 emulators in parallel, changed to generic medium_phone instead and made it configurable. Although these will also change to subsequent PRs as we need to give more control to the user on how the start the android emulator

e.g. mill app.createAndroidDevice <options>